### PR TITLE
Add support for RandomPyPolicy parameter in validate_py_environment

### PR DESCRIPTION
**Pull request by AbhijeetKrishnan**
#584 

Adds an extra parameter to `validate_py_environment`, namely `observation_and_action_constraint_splitter`. This would be useful for environments where randomly sampling the action space produces more invalid actions than valid actions. In such environments, passing an `observation_and_action_constraint_splitter` would make the RandomPyPolicy select only valid actions, potentially making the validation process more thorough.

I have not added any tests to `utils_test.py` and would like help on what some meaningful tests would be.

### CODE REVIEW

File: tf_agents/environments/utils.py [hunk 1]
```diff
@@ -48,14 +49,20 @@ def get_tf_env(
...
+    episodes: int = 5,
+    observation_and_action_constraint_splitter: Optional[
+        types.Splitter] = None):
   """Validates the environment follows the defined specs."""
   time_step_spec = environment.time_step_spec()
   action_spec = environment.action_spec()
 
   random_policy = random_py_policy.RandomPyPolicy(
-      time_step_spec=time_step_spec, action_spec=action_spec)
+      time_step_spec=time_step_spec,
```
Comment (sguada) [inline tf_agents/environments/utils.py]: Remove extra space at the end of the line

File: tf_agents/environments/utils.py [hunk 2]
```diff
@@ -48,14 +49,20 @@ def get_tf_env(
...
   """Validates the environment follows the defined specs."""
   time_step_spec = environment.time_step_spec()
   action_spec = environment.action_spec()
 
   random_policy = random_py_policy.RandomPyPolicy(
-      time_step_spec=time_step_spec, action_spec=action_spec)
+      time_step_spec=time_step_spec, 
+      action_spec=action_spec,
+      observation_and_action_constraint_splitter=(
+        observation_and_action_constraint_splitter))
```
Comment (sguada) [inline tf_agents/environments/utils.py]: indent another 2 spaces.

**Review by sguada — CHANGES_REQUESTED** [review]

**Comment by AbhijeetKrishnan** [comment]
Any follow-up to this?

**Review by kbanoop — APPROVED** [review]

**Comment by kbanoop** [comment]
Thank you. Can you please resolve the conflict? After that I can import it into the internal review system

**Comment by AbhijeetKrishnan** [comment]
Fixed the issue regarding a missing newline at the end of the .gitignore file.

**Pull request merged**